rclpy/test/test_node.py

```diff
@@ -838,6 +838,119 @@ def test_node_set_parameters_rejection(self):
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+
+        self.node.declare_parameters('', reject_list_parameter_tuple1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple1[0][0],
+                    value=reject_list_parameter_tuple1[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[1][0],
+                    value=reject_list_parameter_tuple1[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[2][0],
+                    value=reject_list_parameter_tuple1[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+
+    def test_node_add_on_set_paramter_callback(self):
+        # Add callbacks to the list of callbacks.
+
+        callbacks = [
+            self.reject_parameter_callback,
+            self.reject_parameter_callback_1,
+            self.reject_parameter_callback_2,
+            self.reject_parameter_callback_3
+        ]
+        for callback in callbacks:
+            self.node.add_on_set_parameters_callback(callback)
+
+    def test_node_remove_from_set_callback(self):
+        # Remove callbacks from list of callbacks.
+        # List of callback one of which is rejecting the paramter value to be set.
+
+        paramter_tuple = (
+            'reject1',
+            True,
+            ParameterDescriptor()
+        )
+
+        self.node.declare_parameter(*paramter_tuple)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=paramter_tuple[0],
+                    value=paramter_tuple[0]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+
+        # Removing the callback which is causing the rejection.
+
+        self.node.remove_on_set_parameters_callback(self.reject_parameter_callback_1)
```

Comment:
```suggestion
        self.node.remove_on_set_parameters_callback(self.reject_parameter_callback_1)
        
        self.assertFalse(reject_parameter_callback_1 in self.node._parameters_callbacks)
```

Reply:
@allenh1 sir it looks the same.

Reply:
> what should I change in this?..it looks the same.

There's a new line that asserts that the removed callback is not in the list of callbacks.